sdk/storage/azure-storage-common/inc/azure/storage/common/internal/storage_bearer_token_authentication_policy.hpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <azure/core/http/policies/policy.hpp>
+
+namespace Azure { namespace Storage { namespace _internal {
+
+  class StorageBearerTokenAuthenticationPolicy final
+      : public Core::Http::Policies::_internal::BearerTokenAuthenticationPolicy {
+  public:
+    /**
+     * @brief Construct a Storage Bearer Token challenge authentication policy.
+     *
+     * @param credential An #Azure::Core::TokenCredential to use with this policy.
+     * @param tokenRequestContext A context to get the token in.
+     * @param enableTenantDiscovery Enables tenant discovery through the authorization challenge.
+     */
+    explicit StorageBearerTokenAuthenticationPolicy(
+        std::shared_ptr<Azure::Core::Credentials::TokenCredential const> credential,
+        Azure::Core::Credentials::TokenRequestContext tokenRequestContext,
+        bool enableTenantDiscovery = false)
+        : BearerTokenAuthenticationPolicy(std::move(credential), tokenRequestContext),
+          m_Scopes(tokenRequestContext.Scopes), m_TenantId(tokenRequestContext.TenantId),
+          m_EnableTenantDiscovery(enableTenantDiscovery)
+    {
+    }
+
+    ~StorageBearerTokenAuthenticationPolicy() override {}
+
+    std::unique_ptr<HttpPolicy> Clone() const override
+    {
+      return std::make_unique<StorageBearerTokenAuthenticationPolicy>(*this);
+    }
+
+  private:
+    std::vector<std::string> m_Scopes;
+    mutable std::string m_TenantId;
```

Comment:
why do we need `mutable` here?

Reply:
Because this field may be changed in a [const method](https://github.com/microzchang/azure-sdk-for-cpp/blob/6727a4381ae5e11e4e243a6f9911329dbb696405/sdk/storage/azure-storage-common/src/storage_bearer_token_authentication_policy.cpp#L40)